packages/core/src/command/hl7v2-subscriptions/hl7v2-roster-generator.ts

```diff
@@ -79,7 +80,30 @@ export class Hl7v2RosterGenerator {
     const orgs = await simpleExecuteWithRetries(() => this.getOrganizations([...cxIds]), log);
     const orgsByCxId = _.keyBy(orgs, "cxId");
 
-    const rosterRowInputs = patients.map(p => {
+    const cxsWithAdtsEnabled = await getCxsWithAdtsFeatureFlagEnabled();
+    const adtsEnabledCxSet = new Set(cxsWithAdtsEnabled);
+
+    log(`Checking ${patients.length} patients if they are allowed be in the roster.`);
+    const patientsWithAdtsEnabled: Patient[] = [];
+    for (const patient of patients) {
+      const cxId = patient.cxId;
+
+      // Don't want to call await isAdtsFeatureFla...() here because then we would be making a db call for each patient.
+      if (!adtsEnabledCxSet.has(cxId)) {
+        capture.error(`CX ${cxId} is not allowed to generate roster. Skipping...`, {
+          extra: {
+            cxId: cxId,
+            hieName,
+          },
+        });
+        continue;
+      }
+
+      patientsWithAdtsEnabled.push(patient);
+    }
```

Comment:
This could be done more cleanly - now we know the CXs that should have their patients subscribed, right?

could be done like...

```
    const patients = rawPatients.map(stripInvalidCharactersFromPatientData);

    - const cxIds = new Set(patients.map(p => p.cxId));
    + const cxIds = new Set(await getCxsWithAdtsFeatureFlagEnabled());

    log(`Getting all organizations for patients...`);
    const orgs = await simpleExecuteWithRetries(() => this.getOrganizations([...cxIds]), log);
    const orgsByCxId = _.keyBy(orgs, "cxId");

    ~ const rosterRowInputs = patients.filter(p => cxIds.has(p.cxId)).map(p => {
```